cellar_rebalancer/src/error.rs

```diff
@@ -26,15 +27,24 @@ pub enum ErrorKind {
     /// Gas Oracle error
     #[error("gas error")]
     GasOracle,
+    /// Provider error
+    #[error("grpc error")]
+    GrpcError,
     /// Input/output error
     #[error("http error")]
     Http,
+    /// Cryptographic Keys error
+    #[error("key related error")]
+    KeysError,
+    /// Miscellaneous error
+    ///
+    /// Errors that are returned with types that provide no
+    /// categorical information, such as String
+    #[error("allocation error")]
+    MiscError,
```

Comment:
I don't see any usage of this currently. I don't think this is a useful subtype of error, even if what we're encapsulating is just a string, the context in which it is being returned should provide some useful info for categorizing it. If this is meant to be, as is suggested, an "allocation error", perhaps that is what we mean to call it.

Reply:
I don't remember what this was for either

Reply:
I think this was something I forgot to remove when I realized I didn't need it. There was some function returning a Result<(), String> and I wasn't sure how to handle the String.

Reply:
The method `allocation_precommit` now returns a `Result<somm::AllocationPrecommit, steward::error::Error>, so each error thrown inside need to be able to be converted to `steward::error::Error`. 

On this line, `somm_send::data_hash` result contains an error type `String`, which previously could not be cast as our error type

https://github.com/PeggyJV/steward/blob/a90293af91b969f19f3fc7763ec5117234935dbd/steward/src/allocation.rs#L36-L37

What would be a better way to approach this?

Reply:
The point I was trying to make here would be more that `data_hash` could be returning an error type that categorizes what type of error it is, rather than just wrapping a string in an `Err` response.